src/mnode/mnode-pastel.cpp

```diff
@@ -1571,3 +1571,175 @@ bool CTakeDownTicket::FindTicketInDb(const std::string& key, CTakeDownTicket& ti
 {
     return false;
 }
+
+
+// CChangeUsernameTicket ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+CChangeUsernameTicket CChangeUsernameTicket::Create(std::string _pastelID, std::string _username, const SecureString& strKeyPass)
+{
+    CChangeUsernameTicket ticket(std::move(_pastelID));
+
+    ticket.username = _username;
+
+    // Check if PastelID already have a username on the blockchain. 
+    if (!masterNodeCtrl.masternodeTickets.FindTicketBySecondaryKey(ticket)) {
+        // IF PastelID has no Username yet, the fee is 100 PSL
```

Comment:
it has 100 PSL by default, enough to set 5000 only if need?

Reply:
100 psl first time, 5000 second+ time

Reply:
yep, and you have set 100 PSL as default value in the header. It is not bug. Just repeated code. But if you want to get it more visible here - then it is fine.

Reply:
it is set as default value in Create(), this is is to check if if does match with that default value :)

Reply:
ok ) ticket.fee=100PSL by default so it could be just next
     if (masterNodeCtrl.masternodeTickets.FindTicketBySecondaryKey(ticket)) {
       ticket.fee = masterNodeCtrl.MasternodeUsernameChangeAgainFee;
     }
but it is not problem to keep current code too

Reply:
I do not prefer the idea that we always try to make the code as shorter as possible, while leaving it there make the code easier to understand.